localstack/dns/plugins.py

```diff
@@ -26,3 +26,13 @@ def setup_dns_configuration_on_host():
             server.setup_network_configuration()
     except Exception as e:
         LOG.warning("error setting up dns server: %s", e)
+
+
+@hooks.on_infra_shutdown()
+def stop_server():
```

Comment:
The only issue I have with it, is that we should make sure that we do not do any request after the infra shutdown. All external http requests will probably fail without DNS.

Extensions maybe, among others.  Could we maybe decrease the priority here, so we can make sure we stop the dns after these hooks are executed?